sycl/include/sycl/ext/intel/math/imf_half_trivial.hpp

```diff
@@ -0,0 +1,315 @@
+//==------------- imf_half_trivial.hpp - trivial half utils ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Trival half util functions.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <sycl/half_type.hpp>
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace intel {
+namespace math {
+sycl::half hadd(sycl::half x, sycl::half y) { return x + y; }
+
+sycl::half hadd_sat(sycl::half x, sycl::half y) {
+  return sycl::clamp((x + y), sycl::half(0.f), sycl::half(1.0f));
+}
+
+sycl::half hfma(sycl::half x, sycl::half y, sycl::half z) {
+  return sycl::fma(x, y, z);
+}
+
+sycl::half hfma_sat(sycl::half x, sycl::half y, sycl::half z) {
+  return sycl::clamp(sycl::fma(x, y, z), sycl::half(0.f), sycl::half(1.0f));
+}
+
+sycl::half hmul(sycl::half x, sycl::half y) { return x * y; }
+
+sycl::half hmul_sat(sycl::half x, sycl::half y) {
+  return sycl::clamp((x * y), sycl::half(0.f), sycl::half(1.0f));
+}
+
+sycl::half hneg(sycl::half x) { return -x; }
+
+sycl::half hsub(sycl::half x, sycl::half y) { return x - y; }
+
+sycl::half hsub_sat(sycl::half x, sycl::half y) {
+  return sycl::clamp((x - y), sycl::half(0.f), sycl::half(1.0f));
+}
+
+sycl::half hdiv(sycl::half x, sycl::half y) { return x / y; }
```

Comment:
Could you please also add another trivial function: reciprocal 'hrcp':

sycl::half hrcp(sycl::half x) { return 1.0f / x; }

Reply:
We have already had it and will map hrcp to corresponding _imf_* functions.